lisa/operating_system.py

```diff
@@ -221,15 +409,69 @@ class OpenBSD(BSD):
 
 
 class Fedora(Linux):
+    __fedora_release_pattern_version = re.compile(r"^.*release\s+([0-9\.]+).*$")
+
     @classmethod
     def name_pattern(cls) -> Pattern[str]:
         return re.compile("^Fedora|fedora$")
 
-    def _install_packages(self, packages: Union[List[str]]) -> None:
-        self._node.execute(
-            f"dnf install -y {' '.join(packages)}",
-            sudo=True,
+    def _install_packages(
+        self, packages: Union[List[str]], signed: bool = True
+    ) -> None:
+        command = f"dnf install -y {' '.join(packages)}"
+        if not signed:
+            command.__add__(" --nogpgcheck")
+
+        install_result = self._node.execute(command, sudo=True)
+        if install_result.exit_code != 0:
+            raise LisaException(
+                f"Failed to install {packages}. stdout: {install_result.stdout}"
```

Comment:
1. print the exit_code too
2. make sure the stdout is helpful, and not too much information.